minos/scripts/calculate_cdslen.py

```diff
@@ -14,8 +14,20 @@ def read_fasta(f):
 			yield sid, "".join(seq)
 
 	with open(_out, "w") as fout:
-		for sid, seq in read_fasta(_in):
+		cdna_set = set()
+		for sid, seq in read_fasta(_in_cdna):
+			if sid not in cdna_set:
+				cdna_set.add(sid)
+		cds_set = set()
+		for sid, seq in read_fasta(_in_cds):
+			if sid not in cds_set:
+				cds_set.add(sid)
 			seq = seq.upper().replace("U", "T")
 			has_stop = any(seq.endswith(stop_codon) for stop_codon in ("TAA", "TGA", "TAG"))
 			discard = (has_stop and len(seq) < min_cds_length) or (not has_stop and len(seq) < min_cds_length - 3)
 			print(sid, int(discard), sep="\t", flush=True, file=fout)
+
+		# get the ncRNA models and add them to CDS discard list
+		for sid in cdna_set - cds_set:
+			discard = True
+			print(sid, int(discard), sep="\t", flush=True, file=fout)
```

Comment:
here you can also just do `print(sid, 1, ...)` as the `discard` variable is always `True`

Reply:
I did the way suggested first, but to keep it consistent with the code above I made it explicit.

Reply:
fair enough. it is not like this would cause a performance issue in this case :D